pkg/orchestrator/gitHubActions.go

```diff
@@ -1,79 +1,116 @@
+		return fmt.Errorf("can't parse JSON data: %w", err)
+	}
+	return err
 }
 
 func (g *GitHubActionsConfigProvider) OrchestratorVersion() string {
+	log.Entry().Debugf("OrchestratorVersion() for GitHub Actions is not applicable.")
 	return "n/a"
 }
 
 func (g *GitHubActionsConfigProvider) OrchestratorType() string {
-	return "GitHubActions"
+	return "GitHub Actions"
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildStatus() string {
-	log.Entry().Infof("GetBuildStatus() for GitHub Actions not yet implemented.")
-	return "FAILURE"
+	// By default, we will assume it ass a success
+	// On error it would be handled by the action itself
+	return "SUCCESS"
 }
 
 func (g *GitHubActionsConfigProvider) GetLog() ([]byte, error) {
-	log.Entry().Infof("GetLog() for GitHub Actions not yet implemented.")
+	// It's not possible to get log during workflow execution
+	log.Entry().Debugf("GetLog() for GitHub Actions is not applicable.")
 	return []byte{}, nil
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildID() string {
-	log.Entry().Infof("GetBuildID() for GitHub Actions not yet implemented.")
-	return "n/a"
+	return getEnv("GITHUB_RUN_ID", "n/a")
 }
 
 func (g *GitHubActionsConfigProvider) GetChangeSet() []ChangeSet {
-	log.Entry().Warn("GetChangeSet for GitHubActions not yet implemented")
-	return []ChangeSet{}
+	return []ChangeSet{
+		{
+			g.run.HeadCommit.Id,
+			g.run.HeadCommit.Timestamp.String(),
+			0,
+		},
+	}
 }
 
 func (g *GitHubActionsConfigProvider) GetPipelineStartTime() time.Time {
-	log.Entry().Infof("GetPipelineStartTime() for GitHub Actions not yet implemented.")
-	return time.Time{}.UTC()
+	return g.run.RunStartedAt.UTC()
 }
+
 func (g *GitHubActionsConfigProvider) GetStageName() string {
-	return "GITHUB_WORKFLOW" //TODO: is there something like is "stage" in GH Actions?
+	return getEnv("GITHUB_JOB", "n/a")
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildReason() string {
-	log.Entry().Infof("GetBuildReason() for GitHub Actions not yet implemented.")
-	return "n/a"
+	return getEnv("GITHUB_EVENT_NAME", "n/a")
 }
 
 func (g *GitHubActionsConfigProvider) GetBranch() string {
-	return strings.TrimPrefix(getEnv("GITHUB_REF", "n/a"), "refs/heads/")
+	return getEnv("GITHUB_REF_NAME", "n/a")
 }
 
 func (g *GitHubActionsConfigProvider) GetReference() string {
 	return getEnv("GITHUB_REF", "n/a")
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildURL() string {
-	return g.GetRepoURL() + "/actions/runs/" + getEnv("GITHUB_RUN_ID", "n/a")
+	return g.run.HtmlUrl
 }
 
 func (g *GitHubActionsConfigProvider) GetJobURL() string {
-	log.Entry().Debugf("Not yet implemented.")
-	return g.GetRepoURL() + "/actions/runs/" + getEnv("GITHUB_RUN_ID", "n/a")
+	log.Entry().Infof("GetJobURL() for GitHub Actions is not applicable.")
+	return "n/a"
 }
 
 func (g *GitHubActionsConfigProvider) GetJobName() string {
-	log.Entry().Debugf("GetJobName() for GitHubActions not yet implemented.")
+	log.Entry().Debugf("GetJobName() for GitHub Actions is not applicable.")
```

Comment:
For [Azure DevOps it's returning](https://github.com/SAP/jenkins-library/blob/master/pkg/orchestrator/azureDevOps.go#L85) `<org>/<repo>`, this could also be done here

Reply:
Done